homeassistant/components/influxdb/sensor.py

```diff
@@ -341,3 +441,68 @@ def update(self):
             if len(points) > 1:
                 _LOGGER.warning(QUERY_MULTIPLE_RESULTS_MESSAGE, self.query)
             self.value = points[0].get(INFLUX_CONF_VALUE)
+
+
+class InfluxRawSensorData:
+    """Class for handling the data retrieval with v1 or v2 API and queries_raw format."""
+
+    def __init__(self, influx, api_version, db_name, query, field):
+        """Initialize the data object."""
+        self.influx = influx
+        self.api_version = api_version
+        self.db_name = db_name
+        self.query = query
+        self.field = field
+        self.value = None
+        self.full_query = None
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data with a shell command."""
+        _LOGGER.debug(RENDERING_QUERY_MESSAGE, self.query)
+        try:
+            rendered_query = self.query.render(parse_result=False)
+        except TemplateError as ex:
+            _LOGGER.error(RENDERING_QUERY_ERROR_MESSAGE, ex)
+            return
+
+        self.full_query = f"{rendered_query}"
+
+        _LOGGER.debug(RUNNING_QUERY_MESSAGE, self.full_query)
+
+        if self.api_version == DEFAULT_API_VERSION:
+            try:
+                points = self.influx.query(self.full_query, self.db_name)
+            except (ConnectionError, ValueError) as exc:
+                _LOGGER.error(exc)
+                self.value = None
+                return
+
+            if not points:
+                _LOGGER.warning(QUERY_NO_RESULTS_MESSAGE, self.full_query)
+                self.value = None
+            elif self.field not in points[0]:
+                _LOGGER.warning(QUERY_FIELD_CONSISTENCY, self.field)
+                self.value = None
+            else:
+                if len(points) > 1:
+                    _LOGGER.warning(QUERY_MULTIPLE_RESULTS_MESSAGE, self.full_query)
+                self.value = points[0].get(self.field)
+        else:
+            try:
+                tables = self.influx.query(self.full_query)
+            except (ConnectionError, ValueError) as exc:
+                _LOGGER.error(exc)
+                self.value = None
+                return
+
+            if not tables:
+                _LOGGER.warning(QUERY_NO_RESULTS_MESSAGE, self.full_query)
+                self.value = None
+            elif self.field not in tables[0].records[0].values:
+                _LOGGER.warning(QUERY_FIELD_CONSISTENCY, self.field)
+                self.value = None
+            else:
+                if len(tables) > 1 or len(tables[0].records) > 1:
+                    _LOGGER.warning(QUERY_MULTIPLE_RESULTS_MESSAGE, self.full_query)
+                self.value = tables[0].records[0].values[self.field]
```

Comment:
```suggestion
                self.value = tables[0].records[0].values[INFLUX_CONF_VALUE_V2]
```
Always use `_value` for V2.

Reply:
OK.